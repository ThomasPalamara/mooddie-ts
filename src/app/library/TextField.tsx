import React, { ReactNode } from 'react';
import classNames from 'classnames';
import { TextFieldProps } from './types';

const TextField: React.FunctionComponent<TextFieldProps> = ({
  className,
  label: labelText,
  labelClassName,
  wrapperClassName,
  noWrapper,
  ...other
}) => {
  const classes = classNames(
    'appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500',
    className,
  );
  const wrapperClasses = noWrapper ? classNames('text-field-wrapper mb-5', wrapperClassName) : '';
  const labelClasses = classNames('block text-gray-700 text-sm font-bold mb-3', labelClassName);

  return (
    <div className="wrapperClassName">
      {labelText && <label className={labelClasses}>{labelText}</label>}
      <input className={classes} {...other} />
    </div>
  );
};

TextField.defaultProps = {
  noWrapper: false,
};
export default TextField;
