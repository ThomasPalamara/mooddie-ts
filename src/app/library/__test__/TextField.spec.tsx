import React from 'react';
import { shallow } from 'enzyme';
import { merge } from 'lodash';

import TextField from '../TextField';

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<TextField {...props} />);

  return { wrapper, props };
};
describe('TextField', () => {
  it('should match Snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Label', () => {
  it('should only add a label if label props is passed', () => {
    const { wrapper } = setup({ label: 'Email' });
    expect(wrapper.exists('label')).toBe(true);
    expect(wrapper.find('label').text()).toMatch('Email');
  });
  it('should add custom classes', () => {
    const { wrapper } = setup({ label: 'Name', labelClassName: 'text-red' });
    expect(wrapper.find('label').hasClass('text-red')).toBe(true);
  });
});

describe('Wrapper', () => {
  it('should not render if noWrapper is true', () => {
    const { wrapper } = setup({ noWrapper: true });
    console.log(wrapper.debug());
    expect(
      wrapper
        .find('Wrapper')
        .dive()
        .exists('.text-field-wrapper'),
    ).toBe(false);
  });
  it('should add custom classes', () => {
    const { wrapper } = setup({ wrapperClassName: 'bg-white' });
    expect(
      wrapper
        .find('Wrapper')
        .dive()
        .find('.text-field-wrapper')
        .hasClass('bg-white'),
    ).toBe(true);
  });
});
