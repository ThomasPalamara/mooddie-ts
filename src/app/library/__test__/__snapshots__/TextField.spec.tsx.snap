// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should match Snapshot 1`] = `
<Wrapper>
  <input
    className="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
    type="text"
  />
</Wrapper>
`;
